### PR TITLE
[ISSUE #3909] add domain's judgement

### PR DESCRIPTION
**Pull request by horizonzy**
## What is the purpose of the change
fix-#3909

### CODE REVIEW

File: sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java [hunk 1]
```diff
@@ -219,6 +219,26 @@ public static boolean isIP(String str) {
...
+     */
+    public static boolean isDomain(String str) {
+        try {
+            String[] split = str.split(":");
+            if (split.length != 2) {
+                return false;
+            }
+            Integer.parseInt(split[1]);
+            InetAddress address = InetAddress.getByName(split[0]);
+            return address.isReachable(100);
```
Comment (KomachiSion) [inline sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java:236]: Is this too strict to judge?

this method need the input must can be parse to IP and can reach.

But we just need check the input string is a domain pattern

File: sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java [hunk 2]
```diff
@@ -219,6 +219,26 @@ public static boolean isIP(String str) {
...
     
+    /**
+     * juege str is right domain.
+     *
+     * @param str nacosIp
+     * @return nacosIp is domain
+     */
+    public static boolean isDomain(String str) {
+        try {
+            String[] split = str.split(":");
```
[the remark this answers is not in the document]
Comment (horizonzy) [reply sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java]: yes

File: sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java (hunk 1 above)
Comment (horizonzy) [reply sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java]: yes, just check the domain can resolve

File: sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java [hunk 3]
```diff
@@ -87,7 +88,8 @@ public void run() {
                     nacosIp = ApplicationUtils.getProperty(IP_ADDRESS);
                 }
                 
-                if (!StringUtils.isBlank(nacosIp) && !isIP(nacosIp)) {
+                boolean illegalIp = !StringUtils.isBlank(nacosIp) && !(isIP(nacosIp) || isDomain(nacosIp));
```
Comment (chuntaojun) [inline sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java:91]: illegalIp => illegalIP

File: sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java [hunk 4]
```diff
@@ -219,35 +221,46 @@ public static boolean isIP(String str) {
...
     
+    /**
+     * juege str is right domain.
+     *
+     * @param str nacosIP
+     * @return nacosIP is domain
+     */
+    public static boolean isDomain(String str) {
+        InetSocketAddress address = new InetSocketAddress(str, 0);
+        return !address.isUnresolved();
```
Comment (chuntaojun) [inline sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java:232]: A Warn log can be output here to indicate that the domain name is not reachable

Comment (horizonzy) [reply sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java:232]: solve it

File: sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java [hunk 5]
```diff
@@ -219,35 +221,50 @@ public static boolean isIP(String str) {
...
     
+    /**
+     * juege str is right domain.
+     *
+     * @param str nacosIP
+     * @return nacosIP is domain
+     */
+    public static boolean isDomain(String str) {
+        InetSocketAddress address = new InetSocketAddress(str, 0);
+        boolean isResolved = !address.isUnresolved();
```
Comment (chuntaojun) [inline sys/src/main/java/com/alibaba/nacos/sys/utils/InetUtils.java:232]: isResolved => unResolved

**Review by chuntaojun — APPROVED** [review]

**Pull request merged**